### ISSUE TITLE
PRODUCTION-CRITICAL BACKEND FAILURE INVESTIGATION + SURGICAL FIX

### ISSUE THREAD

**Issue by Anamitra-Sarkar** [Anamitra-Sarkar/chest_disease#8]
FAILURE COST: HIGH (LIVE DEPLOYMENT, USER-FACING ERRORS)  
RISK LEVEL: HIGH (MODEL INFERENCE + EXTERNAL API + FRONTEND CONTRACT)

You are working on a production FastAPI backend deployed on Hugging Face Spaces, consumed by a Vercel-hosted frontend. The system is already live and reachable, but frontend requests intermittently return HTTP 500 errors at runtime.

IMPORTANT:  
You are NOT allowed to redesign, refactor, or “improve” the system.  
This task is UNSAFE if handled approximately.

GOAL  
Eliminate ALL runtime HTTP 500 errors during POST /api/chat requests while preserving exact external behavior and contracts. The backend must behave identically (in stability and error handling) to a known-good previous production system (a heart-disease backend) that used the same frontend integration.

STRICT INVARIANTS (MANDATORY)  
- Exact API routes, request formats, and response schemas MUST remain unchanged  
- Frontend compatibility MUST be preserved exactly  
- No speculative fixes, no “best-effort” handling  
- No silent failure masking  
- Deterministic behavior only  
- Minimal, surgical changes ONLY  
- The system must remain deployable on Hugging Face Spaces with no build or startup regressions  

CONTEXT (DO NOT IGNORE)  
- Backend boots successfully  
- Model loads successfully at startup  
- 500 errors occur ONLY during runtime requests  
- The failing endpoint is POST /api/chat  
- The endpoint supports:
  - message-only requests
  - image-only requests
  - image + message requests
- The backend performs:
  - multipart/form-data parsing
  - image preprocessing
  - PyTorch model inference
  - external LLM API calls
- A previous heart-disease backend (same frontend, same deployment style) does NOT exhibit these 500 errors

REQUIRED WORK (ALL STEPS ARE MANDATORY)  

1. FULL FAILURE SURFACE AUDIT  
   - Enumerate every possible exception path inside POST /api/chat  
   - Explicitly analyze:
     - request parsing
     - missing or empty fields
     - image decoding
     - tensor construction
     - model inference
     - device/dtype issues
     - external LLM API failures
   - No assumptions allowed

2. RUNTIME PATH DIFFERENTIAL ANALYSIS  
   - Compare behavior of:
     a) message-only
     b) image-only
     c) image + message  
   - Ensure NONE of these paths can raise an unhandled exception

3. ERROR HANDLING HARDENING  
   - Replace any generic “catch-all → 500” behavior with:
     - precise logging
     - controlled, deterministic HTTP responses
   - Frontend must NEVER receive an opaque “Failed to process request” without a stable JSON response

4. CONTRACT PRESERVATION  
   - Do NOT change response shape
   - Do NOT add or remove fields
   - Do NOT change HTTP status codes except where strictly required to eliminate crashes

5. DEPLOYMENT SAFETY  
   - Final code must:
     - start cleanly
     - pass runtime requests
     - remain compatible with Hugging Face Spaces
     - not introduce cold-start regressions

DELIVERABLE  
- Output the FULL updated backend file  
- Same structure, same functions, same routes  
- Only stability, validation, and error-handling fixes applied  
- No commentary, no explanations, code only  

ESCALATION CLAUSE  
If you cannot guarantee correctness, determinism, and invariant preservation with your current reasoning capacity, you MUST escalate to a higher-reliability reasoning mode before producing output.

This task is production-critical and unsafe if handled approximately.

Test the backend yourself with this image manually to reverify if the model is able to classify. Here is the image 

<img width="500" height="599" alt="Image" src="https://github.com/user-attachments/assets/b98602b8-e01e-4b93-8afe-f5b5fe1957b6" />

Also check correctly whether the llm is working correctly in the backend and is connected to the frontend 

This is the frontend link 

https://chest-disease.vercel.app/

and here is the link for the embed space of huggingface https://arko007-chest-disease.hf.space

make sure they connect to each other and also I have added the GROQ_API_KEY in the secrets tab in huggingface 

If the model is not able to classify the job is not finished.. 

also update the deployment.md according to this deployment for backend in huggingface and frontend in vercel about the correct deployment steps and env vars 

double check everything before finalising 

currently these are the backend logs on huggingface .. 

Fix all of them 

===== Application Startup at 2026-01-22 14:04:15 =====

INFO:     Started server process [7]
INFO:     Waiting for application startup.
INFO:backend.main:Starting up Chest X-Ray Analysis API
INFO:backend.main:Loading model from: epoch_001_mAUROC_0.486525.pth
INFO:backend.main:Loaded model from checkpoint (ema_state_dict, epoch: 1)
INFO:backend.main:Model loaded successfully
INFO:backend.main:API ready to serve requests
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:7860 (Press CTRL+C to quit)
INFO:     10.16.30.4:24931 - "GET / HTTP/1.1" 404 Not Found
ERROR:backend.main:Chat error: Client.__init__() got an unexpected keyword argument 'proxies'
ERROR:backend.main:Chat endpoint error: Failed to process chat: Client.__init__() got an unexpected keyword argument 'proxies'
INFO:     10.16.0.113:26585 - "POST /api/chat HTTP/1.1" 500 Internal Server Error
INFO:     10.16.30.4:3967 - "HEAD /health HTTP/1.1" 405 Method Not Allowed
